### PR TITLE
Fix for libpbs.so linking with standalone code using PBS IFL API

### PR DESCRIPTION
**Pull request by hirenvadalia**
#### Describe Bug or Feature

While linking with libpbs.so it fails with missing symbols error because respective sources files are not listed in the libpbs sources list.

#### Describe Your Change

Add missing files in the libpbs sources list.

#### Link to Design Doc

None

#### Attach Test and Valgrind Logs/Output

[before.txt](https://github.com/PBSPro/pbspro/files/4629906/before.txt)
[after.txt](https://github.com/PBSPro/pbspro/files/4629971/after.txt)

### CODE REVIEW

File: test/tests/interfaces/pbs_libpbs_so.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+from tests.interfaces import *
+
+
+class TestLibpbsLinking(TestInterfaces):
+    """
+    Test suite to shared libpbs library linking
+    """
+
+    @tags('smoke')
```
Comment (bhroam) [inline test/tests/interfaces/pbs_libpbs_so.py:46]: Do we really need to worry about someone linking against libpbs as a smoke test?

File: test/tests/interfaces/pbs_libpbs_so.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        self.assertEqual(_res['rc'], 0)
+        self.assertEqual(len(_res['out']), 2)
+        _s = self.server.status(SERVER)[0]
+        _exp = ["%s = %s" % (ATTR_SvrHost, _s[ATTR_SvrHost])]
+        _exp += ["%s = %s" % (ATTR_total, _s[ATTR_total])]
+        _exp = "\n".join(_exp)
+        _out = "\n".join(_res['out'])
+        self.assertEqual(_out, _exp)
+
+test_code = '''
```
Comment (bhroam) [inline test/tests/interfaces/pbs_libpbs_so.py:84]: Move this variable up ahead of the test.

File: test/tests/interfaces/pbs_libpbs_so.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        a = p->attribs;
+        while (a != NULL) {
+            if (a->name != NULL &&
+                (!strcmp(a->name, ATTR_SvrHost) ||
+                    !strcmp(a->name, ATTR_total))) {
+                printf("%s = %s\\n", a->name, a->value);
+            }
+            a = a->next;
+        }
+        p = p->next;
```
Comment (bhroam) [inline test/tests/interfaces/pbs_libpbs_so.py:112]: Currently a statserver returns 1 batch_status.  I don't think this is changing with multiserver.  I don't think you need this outer loop.

**Comment by hirenvadalia** [comment]
@bhroam Can you please review the latest changes? This is a kinda blocker issue for the next release. Thank you!

**Comment by hirenvadalia** [comment]
@PBSPro/pbspro-maintainers Can I get one more review for this PR, please?

**Comment by bhroam** [comment]
@hirenvadalia you have a maintainer review.  The second review should come from your team.

**Comment by hirenvadalia** [comment]
> @hirenvadalia you have a maintainer review. The second review should come from your team.

Yes, @bhroam forget that! I just had chat with @agrawalravi90 he is doing the review.

**Pull request merged**